src/semigroup/object.js

```diff
@@ -1,11 +1,29 @@
 import { Semigroup } from '../semigroup';
 import propertiesOf from 'object.getownpropertydescriptors';
+import stable, { Stable } from '../stable';
 
-const { assign, getPrototypeOf } = Object;
+const { assign, getPrototypeOf, keys } = Object;
 
 Semigroup.instance(Object, {
   append(o1, o2) {
     let properties = assign({}, propertiesOf(o1), propertiesOf(o2));
-    return Object.create(getPrototypeOf(o1), properties);
+    return Object.create(getPrototypeOf(o1), cacheGetters(properties));
   }
-});
\ No newline at end of file
+});
+
+function cacheGetters(descriptors) {
+  return keys(descriptors).reduce(function(memo, key) {
+    let descriptor = descriptors[key];
+    if (descriptor.get && descriptor.get[Stable] || !descriptor.get) {
```

Comment:
Let's just put the check for `Stable` inside the `stable` function. 

That way, we can test it with:

```js
const stablelized = stable(function() {});
stable(stableized) === stableized //=> true
```

And then consumers don't need to worry about accidentally double-wrapping a function at all.

Reply:
Done